src/core/utils/hooks/useOverflow.tsx

```diff
@@ -6,6 +6,8 @@ import React from 'react';
 import { useMergedRefs } from './useMergedRefs';
 import { useResizeObserver } from './useResizeObserver';
 
+const STARTING_ITEMS_COUNT = 20;
```

Comment:
Starting by rendering 20 items (just kind of a random number) because rendering all of the items e.g. 2.4k is too much and it becomes laggy.

Reply:
Maybe should have "MAX" in the name, as it's the maximum (items.length will be used if its lower).

Also maybe make it a power of 2 since we are multiplying by 2 repeatedly 😆 so 16 would be a nice value

Reply:
For me it is easier to work with numbers that divide by 10, maybe I am not a real programmer 😕

Reply:
Added `MAX`. He is a really great guy!